sources/api/shibaken/src/warmpool/config.rs

```diff
@@ -0,0 +1,58 @@
+use crate::warmpool::error::{self, Result};
+use serde::Deserialize;
+use snafu::ResultExt;
+use std::fs;
+use std::path::Path;
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct Config {
+    pub(crate) should_wait: bool,
+    pub(crate) marker_path: String,
+}
+
+impl Config {
+    /// Parses configuration file at passed in path and returns struct containing settings value.
+    pub(crate) fn from_file<P: AsRef<Path>>(path: P) -> Result<Self> {
+        let path = path.as_ref();
+        let string_parse = fs::read_to_string(path).context(error::ConfigReadSnafu { path })?;
+        let config: Config =
+            toml::from_str(&string_parse).context(error::ConfigParseSnafu { path })?;
+        Ok(config)
+    }
+}
```

Comment:
I suppose it's a matter of taste but this is a pretty small module - I would personally just include the struct and tests in the same module where it's used.